frontend/src/pages/search.tsx

```diff
@@ -1,84 +1,218 @@
-          { offset: 0, limit: 5, tags: [], groups: [], orgs: [] },
-        ])]: search_result,
-      },
-      groups,
-      tags,
-      orgs,
-    },
+    props: query,
   };
 }
 
+export type SearchPageOnChange = (
+  data: {
+    value: string[] | boolean | string | number | undefined;
+    key: keyof SearchDatasetType;
+  }[]
+) => void;
+
 export default function DatasetSearch({
-  fallback,
-  groups,
-  tags,
-  orgs,
-}: InferGetServerSidePropsType<typeof getStaticProps>): JSX.Element {
-  const router = useRouter();
-  const { q, sector, mode, service, region } = router.query;
-  const [options, setOptions] = useState<PackageSearchOptions>({
+  query,
+  sector,
+  mode,
+  service,
+  region,
+  fuel,
+  before,
+  after,
+  country,
+}: any) {
+  const [modes, setModes] = useState<Facet[]>([]);
+  const [services, setServices] = useState<Facet[]>([]);
+  const [updateFrequencies, setUpdateFrequencies] = useState<Facet[]>([]);
+  const [tags, setTags] = useState<Facet[]>([]);
+  const [sectors, setSectors] = useState<Facet[]>([]);
+  const [orgs, setOrgs] = useState<Facet[]>([]);
+  const [resourcesFormats, setResourcesFormats] = useState<Facet[]>([]);
+  const [regions, setRegions] = useState<Facet[]>([]);
+  const [countries, setCountries] = useState<Facet[]>([]);
+  const [metadataCreatedDates, setMetadataCreatedDates] = useState<Facet[]>([]);
+  const [yearsCoverage, setYearsCoverage] = useState<Facet[]>([]);
+  const [showAdvancedFilter, setShowAdvancedFilter] = useState(false);
+
+  const resetFilter = () => {
+    setSearchFilter({
+      offset: 0,
+      limit: 9,
+      sort: "score desc, metadata_modified desc",
+      facetsFields: `["tags", "groups", "services", "modes", "sectors","frequency","regions", "geographies", "organization", "res_format", "metadata_created"]`,
+    });
+    setCurrentPage(0);
+  };
+
+  const [searchFilter, setSearchFilter] = useState<SearchDatasetType>({
     offset: 0,
-    limit: 5,
-    tags: [],
-    groups: [],
-    orgs: [],
-    query: q as string,
+    limit: 9,
+    endYear: before ? Number(after) : undefined,
+    startYear: after ? Number(after) : undefined,
+    modes: mode ? [mode as string] : undefined,
+    services: service ? [service as string] : undefined,
+    sectors: sector ? [sector as string] : undefined,
+    fuel: fuel as string | undefined,
+    regions: region ? [region as string] : undefined,
+    countries: country ? [country as string] : undefined,
+    query: query as string,
+    sort: "score desc, metadata_modified desc",
+    facetsFields: `["tags", "groups", "services", "modes", "sectors","frequency","regions", "geographies", "organization", "res_format", "metadata_created"]`,
   });
 
+  const [currentPage, setCurrentPage] = useState(0);
+  const {
+    data: { datasets, count: datasetCount, facets } = {
+      datasets: [],
+      facets: {} as any,
+    },
+  } = api.dataset.search.useQuery(searchFilter);
+
+  useEffect(() => {
```

Comment:
Same thing as before, mostly a suggestion for you to read, https://react.dev/learn/you-might-not-need-an-effect if all your useEffect does is call setState, then its probably not needed

Reply:
in this case, if I didn't add a useEffect it entered in a loop, because in this for it's changing the states of the page, and the page reloaded because states were changed entering an infinite loop.